src/Parsers/ExpressionListParsers.cpp

```diff
@@ -221,6 +221,7 @@ static bool modifyAST(ASTPtr ast, SubqueryFunctionType type)
     select_with_union_query->list_of_selects = std::make_shared<ASTExpressionList>();
     select_with_union_query->list_of_selects->children.push_back(std::move(select_query));
     select_with_union_query->children.push_back(select_with_union_query->list_of_selects);
+    select_with_union_query->setHasQueryParameters();
```

Comment:
imo it would be better to just set it on first attempt to get it, further attempts will use already set value (you can use std::optional for that). (because parsers should just parse and not execute any additional logic)

Reply:
I considered this approach, but would have to make getter `hasQueryParameters()` not const. In places it is used, its called by const query, and I too think its good to keep this function const.
Please let me know if you have any suggestions on this.

Reply:
may be just use `mutable` (for the field that will be changed)? then the problem will be solved

Reply:
addressed it, thanks